crates/compiler/builtins/roc/Num.roc

```diff
@@ -867,6 +868,35 @@ max = \a, b ->
     else
         b
 
+## Restricts a number to be between two other numbers.
+##
+## ```
+## restrictToInterval 5 3 8 == 5
+## restrictToInterval 1 3 8 == 3
+## restrictToInterval 10 3 8 == 8
+##
+## restrictToInterval 5 8 3 == 5
+## restrictToInterval 1 8 3 == 3
+## restrictToInterval 10 8 3 == 8
+## ```
+restrictToInterval : Num a, Num a, Num a -> Num a
```

Comment:
I'd like to use something like the record with `start` and `end` like we do with `List.range`. I'll discuss on zulip.

Reply:
https://roc.zulipchat.com/#narrow/stream/304641-ideas/topic/Name.20and.20Type.20signature.20for.20Num.2Eclamp/near/408335035